larpandoracontent/LArHelpers/LArMCParticleHelper.cc

```diff
@@ -388,11 +549,20 @@ void LArMCParticleHelper::GetPfoMCParticleHitSharingMaps(const PfoContributionMa
 //------------------------------------------------------------------------------------------------------------------------------------------
 
 void LArMCParticleHelper::CollectReconstructable2DHits(const ParticleFlowObject *const pPfo, const MCContributionMapVector &selectedMCParticleToHitsMaps,
-    pandora::CaloHitList &reconstructableCaloHitList2D)
+    pandora::CaloHitList &reconstructableCaloHitList2D, const bool testBeamHierarchyMode)
 {
     // Collect all 2D calo hits in pfo hierarchy
     PfoList pfoList;
-    LArPfoHelper::GetAllDownstreamPfos(pPfo, pfoList);
+
+    // ATTN: Only collect downstream pfos for daughter test beam particles, cosmics, neutrinos
+    if (testBeamHierarchyMode && pPfo->GetParentPfoList().empty() && LArPfoHelper::IsTestBeam(pPfo))
```

Comment:
Suggest create a function in the LArPfoHelper to sit alongside the `GetAllDownstreamPfos` function, but for this use case. In this way, takes away the need for any actual logic (no matter how small/trivial) hidden inside this function.